### PR TITLE
More dewikify fixes

### PR DESCRIPTION
**Pull request by matthijskooijman**
This fixes a bunch more problems with (de)wikifying. Turned out that while wikifying was recently fixed, the period of it being broken had caused some bitrot with the dewikifying, which was no longer working or properly being done in a bunch of places. This should fix all of them, by more thoroughly running dewikify in a few general places.

### CODE REVIEW

**Comment by matthijskooijman** [comment]
I justed added one more commit that makes all urls in outgoing e-mails absolute. Not entirely on-topic, but close enough (and we needed that for Meetjestad just now). That commit hasn't been tested thoroughly yet.

**Comment by matthijskooijman** [comment]
I've just updated this PR to provide a better fix for text-handling in links by dewikify (backporting a change from upstream php-dom-wrapper, and removing two commits that added workarounds in dewikify) and added wordwrapping for outgoing e-mail (still a bit rough, but better than nothing).

File: system/core/pages.php [hunk 1]
```diff
@@ -306,6 +306,21 @@ function showPagename($name) {
 		return preg_replace('/_/', ' ', $name);
 	}
 
+	/* Like dewikify, but accepts a HTML string instead of a document */
+	function dewikify_html($html) {
+		$doc = new DomWrap\Document();
+		// Dewikify a separate element instead of the entire
+		// document, since loading html into a document adds
+		// <html><body>.
+		$elem = $doc->createElement('root');
```
[the remark this answers is not in the document]
Comment (matthijskooijman) [reply system/core/pages.php]: What do you mean by destroyed? PHP does reference counting / garbage collection, so it should clean up things by itself? Everything happens in a new document, which should also be cleaned up?

[the remark this answers is not in the document]
Comment (laurensmartina) [reply system/core/pages.php]: I somehow did not notice that `$doc` is a new instance, created within this method. So `$doc` gets destroyed directly after exiting the method and with it so does the newly created element.
In sort, ignore my comment here. :-)

[the remark this answers is not in the document]
Comment (matthijskooijman) [reply system/core/pages.php]: Ok, ignored ;-)

Also, note that `createElement` *creates* a node, but does not *add* it anywhere in the tree.

**Pull request merged**